src/test/java/unit/GcsBucketLifecycle.java

```diff
@@ -0,0 +1,390 @@
+  }
+
+  @Test
+  @DisplayName("multiple lifecycle rules")
+  void multipleRules() throws IOException {
+    String name = "multipleRules";
+    List<? extends BucketInfo.LifecycleRule> lifecycleRules =
+        createBucketWithLifecycleRules(name, name + ".json");
+    validateMultipleRules(lifecycleRules);
+  }
+
+  @Test
+  @DisplayName("CLI uses the same format as gsutil for setting lifecycle rules")
+  void sameFormatForExternalBucket() throws IOException {
+    // the CLI mounts the current working directory to the Docker container when running apps
+    // so we need to give it the path to lifecycle JSON file relative to the current working
+    // directory
+    Path lifecyclePathOnHost = TestCommand.getPathForTestInput("gcslifecycle/multipleRules.json");
+    Path currentDirOnHost = Path.of(System.getProperty("user.dir"));
+    Path lifecycelPathOnContainer = currentDirOnHost.relativize(lifecyclePathOnHost);
+
+    // `terra gsutil lifecycle set $lifecycle gs://$bucketname`
+    TestCommand.runCommandExpectSuccess(
+        "gsutil",
+        "lifecycle",
+        "set",
+        lifecycelPathOnContainer.toString(),
+        "gs://" + externalBucket.getName());
+
+    List<? extends BucketInfo.LifecycleRule> lifecycleRules =
+        getLifecycleRulesFromCloud(externalBucket.getName());
+    validateMultipleRules(lifecycleRules);
+  }
+
+  /**
+   * Assert that the bucket lifecycle rules retrieved from GCS directly match what's expected for
+   * the multipleRules.json file.
+   */
+  private void validateMultipleRules(List<? extends BucketInfo.LifecycleRule> lifecycleRules)
+      throws IOException {
+    assertEquals(2, lifecycleRules.size(), "bucket has two lifecycle rules defined");
+
+    Optional<? extends BucketInfo.LifecycleRule> ruleWithDeleteAction =
+        lifecycleRules.stream()
+            .filter(rule -> rule.getAction().getActionType().equals("Delete"))
+            .findFirst();
+    assertTrue(ruleWithDeleteAction.isPresent(), "one rule has action type = delete");
+    expectActionDelete(ruleWithDeleteAction.get());
+    assertEquals(84, ruleWithDeleteAction.get().getCondition().getAge(), "condition age matches");
+
+    Optional<? extends BucketInfo.LifecycleRule> ruleWithSetStorageClassAction =
+        lifecycleRules.stream()
+            .filter(rule -> rule.getAction().getActionType().equals("SetStorageClass"))
+            .findFirst();
+    assertTrue(
+        ruleWithSetStorageClassAction.isPresent(), "one rule has action type = set storage class");
+    expectActionSetStorageClass(ruleWithSetStorageClassAction.get(), StorageClass.COLDLINE);
+    assertFalse(
+        ruleWithSetStorageClassAction.get().getCondition().getIsLive(),
+        "condition is live matches");
+  }
+
+  /** Check that the action is Delete. */
+  private void expectActionDelete(BucketInfo.LifecycleRule rule) {
+    assertEquals("Delete", rule.getAction().getActionType(), "Delete action type matches");
```

Comment:
You can use `DeleteLifecycleAction.TYPE` or some such for this.

Reply:
done.